datatypes/src/primitives/query_rectangle.rs

```diff
@@ -38,3 +38,48 @@ impl From<QueryRectangle<BoundingBox2D>> for QueryRectangle<SpatialPartition2D>
         }
     }
 }
+
+/// manual implementation, because derivation can't handle the `SpatialBounds` generic (yet)
+impl utoipa::Component for QueryRectangle<SpatialPartition2D> {
```

Comment:
is it reasonable to use utoipa in the datatypes crate?
Or does this reflect a bad design of us? Actually, it should only be part of services?

Reply:
it's definitely suboptimal that our API data types are scattered the way they are. However, I am not sure if we should really duplicate everything we need in services. I guess we have to discuss this.